### PR TITLE
Fix XBlockAside.add_xml_to_node crash

### PR DESCRIPTION
**Pull request by StraNNiKK**
This PR is related with the conversation: https://github.com/edx/edx-platform/pull/11808#discussion_r56330814

`has_children` attribute is set in [ChildrenModelMetaclass](https://github.com/edx/XBlock/blob/master/xblock/mixins.py#L333) metaclass. `ChildrenModelMetaclass` is metaclass for [HierarchyMixin](https://github.com/edx/XBlock/blob/master/xblock/mixins.py#L355). `XBlock` inherits `HierarchyMixin`:
```
class XBlock(XmlSerializationMixin, HierarchyMixin, ScopedStorageMixin,
RuntimeServicesMixin, HandlersMixin, IndexInfoMixin, ViewsMixin, SharedBlockBase):
```
but `XBlockAside` isn't:
```
class XBlockAside(XmlSerializationMixin, ScopedStorageMixin, 
RuntimeServicesMixin, HandlersMixin, SharedBlockBase):
```
The `self.has_children` check is used in `XmlSerializationMixin.add_xml_to_node` function.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @StraNNiKK! I've created [OSPR-1200](https://openedx.atlassian.net/browse/OSPR-1200) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by cpennington** [comment]
:+1:

**Comment by cpennington** [comment]
Thanks for fixing this!

**Comment by jbarciauskas** [comment]
@cpennington do we just need to press the green button?

File: xblock/test/django/test_request.py [hunk 1]
```diff
@@ -80,12 +80,12 @@ def test_headers(self):
     def test_content_type(self):
         # Default charset
         self.assertEquals(
-            self._as_django(content_type='application/json')['Content-Type'],
-            'application/json; charset=UTF-8'
+            self._as_django(content_type='text/html')['Content-Type'],
+            'text/html; charset=UTF-8'
```
Comment (nedbat) [inline xblock/test/django/test_request.py]: I'm concerned about these test changes.  It seems to be that we are moving from a test that behaves similar to real code, to a test that behaves less like real code.  PR #338 has much more discussion about it.

Comment (StraNNiKK) [inline xblock/test/django/test_request.py]: @nedbat ok. Do I need to revert these changes?

**Comment by mattdrayer** [comment]
Okay, so based on the context in #338, where are we at now with this PR?  Sorry to be such a pain about it, but I really would like to see #335 merge soon.

**Comment by mattdrayer** [comment]
Checking back in on this PR -- where are we at with it?

**Comment by cpennington** [comment]
@mattdrayer how is this related to #335?

**Comment by cpennington** [comment]
I thought that this was targeted at fixing a bug found in https://github.com/edx/edx-platform/pull/11808

File: xblock/test/django/test_request.py (hunk 1 above)
Comment (cpennington) [inline xblock/test/django/test_request.py]: @nedbat: All this test was intending to test was the behavior of the `charset` argument to `_as_django`. I think the main issue with the test as it was written as that it assumed it knew what charset webob was going to provide by default, and thus it broke when webob changed that default. I think the test should just be changed to assert that the content-type is set to the supplied content type, but ignore the charset, and then if the charset is set (as below), verify that `_as_django` sets the charset correctly.

**Comment by mattdrayer** [comment]
I was told to hold off on merging #335 until #337 merged, because it fixes the issue being discussed in #338 (although this PR doesn't appear to address the problem correctly).  IMHO neither #337 nor #338 actually block #335, so I would like to just merge it, but I'm trying to work with peoples' concerns.

**Comment by cpennington** [comment]
Yeah, #335 seems unrelated to the charset issues to me, except that its suffering from breakage from an unpinned webob version. If we want to have tests pass, then pinning webob to the current version (that the code is known to work with) in #335, and then upgrading webob in a separate PR is probably the way to go (at least in my mind).

**Comment by mattdrayer** [comment]
If we *must* see green on #335 then I am happy to pin to webob v1.5.1 so we can move forward.  We had done this at one point but removed the pin after some discussion.

**Comment by nedbat** [comment]
Is there a reason it's hard to fix #338 the right way and do the right thing everywhere?

**Comment by mattdrayer** [comment]
I'm still not clear on what the 'right way' is?

**Comment by mattdrayer** [comment]
OKAY, after discussing further with @nedbat we determined what needed to be done with PR #338 and that branch has been updated and merged -- a rebase is now in order for this change set, which should address the WebOb-related test failure.

**Comment by StraNNiKK** [comment]
@mattdrayer @nedbat @cpennington hi. I've made rebase and updated the PR

**Comment by jbarciauskas** [comment]
@mattdrayer @nedbat @cpennington is this ready to go?
